### DISCUSSION TITLE
Is there a way to get Appium to attempt to uninstall a driver 'only if' they exist?

### DISCUSSION THREAD

**Discussion by tking16** [appium/appium#16598] [Q&A] [answered]
Our Appium framework is handled by a Makefile which installs on a machine after every deployment. I'm struggling a bit with adding the appium driver install and uninstall commands for 2.0. We already have a `clean` command that wipes all the node_modules, when I added `node $(APPIUM_PATH) driver uninstall xcuitest` and  `node $(APPIUM_PATH) driver uninstall uiautomator2` the command would error if they weren't installed. My question is, is there a way for the command to carry on gracefully should it not find the appium drivers?

I also have some trouble installing via the Makefile, but don't want to make this question too long.

**Comment by RSM-SBA** [comment]
on mac you can pipe a true

appium driver uninstall xcuitests || true

**Reply by shylendramadda** [reply]
It worked for me thank you

**Comment by jlipps** [comment, marked as answer]
With the new version of the Appium 2 beta you can also just manage dependencies as node modules in your project and not use the Appium CLI to manage them. I.e., you can just put `appium` and `appium-uiautomator2-driver` in your package.json and Appium should be able to find the driver. Then they can both be managed by your existing clean logic without having to worry about cleaning APPIUM_HOME as well.

**Reply by tking16** [reply]
Thanks @jlipps, does this count for installing too? So would I be able to simply add my required drivers to package.json or would I always need to run `appium driver install xcuitest` for example? At the moment my drivers aren't being found even though they're in my package.json and I'm running `npm install`.

**Reply by jlipps** [reply]
It could be that the version of Appium 2 which supports this functionality has not yet been published as a beta. I'll give it a look.

**Reply by jlipps** [reply]
OK, I confirmed that beta 25 did not have this feature yet, so I published beta 26 just now. Give it a try. And yes, you should be able to just put appium itself and any drivers as NPM dependencies, run `npm install` in that dir, and then `npx appium` should pick up the drivers that are managed by your package.json.

**Reply by tking16** [reply]
Sorry took a while to get back. Thanks I just tried and it worked!
